core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -146,294 +82,69 @@ export class ResultPanelComponent {
+
     const executionState = this.executeWorkflowService.getExecutionState();
 
     if (executionState.state === ExecutionState.Failed) {
-      this.errorMessages = this.executeWorkflowService.getErrorMessages();
+      this.component = ConsoleFrameComponent;
     } else if (executionState.state === ExecutionState.BreakpointTriggered) {
-      const breakpointTriggerInfo = this.executeWorkflowService.getBreakpointTriggerInfo();
-      if (this.resultPanelOperatorID && this.resultPanelOperatorID === breakpointTriggerInfo?.operatorID) {
-        this.breakpointTriggerInfo = breakpointTriggerInfo;
-        this.breakpointAction = true;
-        const result = breakpointTriggerInfo.report.map(r => r.faultedTuple.tuple).filter(t => t !== undefined);
-        this.setupResultTable(result, result.length);
-        const errorsMessages: Record<string, string> = {};
-        breakpointTriggerInfo.report.forEach(r => {
-          const pathsplitted = r.actorPath.split('/');
-          const workerName = pathsplitted[pathsplitted.length - 1];
-          const workerText = 'Worker ' + workerName + ':                ';
-          if (r.messages.toString().toLowerCase().includes('exception')) {
-            errorsMessages[workerText] = r.messages.toString();
-          }
-        });
-        this.errorMessages = errorsMessages;
-      }
+      this.component = ConsoleFrameComponent;
     } else {
-      // display result table by default
       if (this.resultPanelOperatorID) {
-        const paginatedResultService = this.workflowResultService.getPaginatedResultService(this.resultPanelOperatorID);
         const resultService = this.workflowResultService.getResultService(this.resultPanelOperatorID);
-
-        this.resultUpdateSubscription = this.workflowResultService.getResultUpdateStream().subscribe(update => {
-          if (! this.resultPanelOperatorID) {
-            return;
-          }
-          const opUpdate = update[this.resultPanelOperatorID];
-          if (! opUpdate || ! isWebPaginationUpdate(opUpdate)) {
-            return;
-          }
-          this.totalNumTuples = opUpdate.totalNumTuples;
-          this.isFrontPagination = false;
-          if (opUpdate.dirtyPageIndices.includes(this.currentPageIndex)) {
-            this.changePaginatedResultData();
-          }
-        });
-
+        const paginatedResultService = this.workflowResultService.getPaginatedResultService(this.resultPanelOperatorID);
         if (paginatedResultService) {
-          this.isFrontPagination = false;
-          this.totalNumTuples = paginatedResultService.getCurrentTotalNumTuples();
-          this.currentPageIndex = paginatedResultService.getCurrentPageIndex();
-          this.changePaginatedResultData();
-
+          this.component = ResultTableFrameComponent;
         } else if (resultService && resultService.getChartType()) {
-          this.chartType = resultService.getChartType();
+          this.component = VisualizationFrameComponent;
```

Comment:
I've added `this.workflowResultService.getResultUpdateStream()` as one of the trigger events, so that when the first `result` is coming, it will trigger rerender.

Reply:
changed to `this.workflowResultService.getResultInitiateStream()` to reduce events